### PR TITLE
Improvement/Keychain cancelled issue

### PR DESCRIPTION
**Pull request by vignesha22**
Added Regex as discussed for neglecting the error caused by user when the biometric is cancelled

### CODE REVIEW

File: src/utils/keychain.js [hunk 1]
```diff
@@ -48,38 +48,41 @@ export type KeyChainData = {
...
-  const colors = getThemeColors();
-  const buttons = [];
-  buttons.push({
-    text: t('error.failedKeychain.exitButtonText'),
-    onPress: () => RNExitApp.exitApp(),
-  });
-  accountAddress = accountAddress ?? etherspotService?.getAccountAddress(CHAIN.ETHEREUM);
-  if (accountAddress) {
+export const handleCatch = (accountAddress: ?string, error: ?any[]) => {
+  const msg = error ? error[0]?.Error?.msg : '';
```
Comment (IAmKio) [inline src/utils/keychain.js:52]: Could you just `.toString()` `error` here?

File: src/utils/keychain.js [hunk 2]
```diff
@@ -48,38 +48,41 @@ export type KeyChainData = {
...
+  if (msg && !/cancel/gi.exec(msg)) {
+    reportErrorLog('Exception caught on keychain: ', error);
```
Comment (IAmKio) [inline src/utils/keychain.js:54]: Please move the error reporting outside the `if` statement, we want to report all errors even if they are just actions that are cancelled.

**Review by IAmKio — CHANGES_REQUESTED** [review]
Just some minor changes

File: src/utils/keychain.js (hunk 2 above)
Comment (vignesha22) [reply src/utils/keychain.js:54]: okay

File: src/utils/keychain.js (hunk 1 above)
Comment (vignesha22) [reply src/utils/keychain.js:52]: that's what I thought but it won't return as json to separate code and then have to parseJson and then take msg

**Review by IAmKio — APPROVED** [review]
LGTM

**Pull request merged**